pkgs/applications/virtualization/docker/default.nix

```diff
@@ -13,11 +11,14 @@ rec {
       , stdenv, fetchFromGitHub, fetchpatch, buildGoPackage
       , makeWrapper, installShellFiles, pkg-config, glibc
       , go-md2man, go, containerd, runc, docker-proxy, tini, libtool
-      , sqlite, iproute2, lvm2, systemd, docker-buildx, docker-compose
-      , btrfs-progs, iptables, e2fsprogs, xz, util-linux, xfsprogs, git
-      , procps, libseccomp, rootlesskit, slirp4netns, fuse-overlayfs
-      , nixosTests
+      , sqlite, iproute2, docker-buildx, docker-compose
+      , iptables, e2fsprogs, xz, util-linux, xfsprogs, git
+      , procps, rootlesskit, slirp4netns, fuse-overlayfs, nixosTests
       , clientOnly ? !stdenv.isLinux, symlinkJoin
+      , withSystemd ? true, systemd
+      , withBtrfs ? true, btrfs-progs
+      , withLvm ? true, lvm2
```

Comment:
This made docker unsupported on darwin:

* https://hydra.nixos.org/eval/1786368#tabs-removed

Reply:
Same applies to `withSystemd`, `withBtrfs` and `withSeccomp`.

Reply:
#203535